darts/models/forecasting/nlinear.py

```diff
@@ -129,7 +129,7 @@ def forward(
             x = self.layer(x)  # (batch, out_dim, out_len * nr_params)
 
             if self.normalize:
-                x = x + seq_last
+                x = x + seq_last[:, :, : self.nr_params]  # (batch, out_dim, nr_params)
```

Comment:
Is this different from the current implementation?

From line 126 `seq_last` will have shape (batch, out_dim, 1).  So even if we take `seq_last[:, :, :self.nr_params]` here, it will still have shape (batch, out_dim,1).

Maybe I'm missing something?

Reply:
Its specifically for the case when you have covariates. for example if you have 2 time series your predicting for and 3 covariates:

the linear layer will output (B , T, 2 ) but seq last will be of shape (B, T, 5) which will error because the C dim does not match in shape. This fix is specifically to account for that.

Reply:
Could you have a look at line 126 which defines `seq_last` (see below)? It will always be of shape (B, T, 1) no matter what input we have:

```
seq_last = x[:, :, -1:].detach()  # (batch, out_dim, 1)
```

Reply:
Your right for the case when self.shared_weights is true. It looks like when `self.shared_weights` is true **the code simply disregards the covariates**. 

line 122:
            ```# discard covariates, to ensure that in_dim == out_dim
            x = x[:, :, : self.output_dim]```

I didn't realize that the behavior was different for the shared weights. 


looking back, I misunderstood nr_params at the time (didnt realize it was for likelihood forecasts). I encountered the issue in the case when nr_params == 1

For when `self.shared_weights` is false, the on line 150 was the issue I encountered.

in this case, `seq_last` is of shape `Batch, 1 , in_dim` but the output from the linear layer `x` is of shape `Batch, Timesteps, out_dim * nr_params` 

I'm realizing now that in this case the correct behavior would be to reshape `x` to `(Batch, Timestep, n_params, out_dim` and seq_last as `(Batch, 1, 1, :out_dim)` and do subtraction that way.

If the correct behavior for when `shared_weights` is false is to disregard past covariates, I'd be happy to update my PR to disallow past covariates, as well as update the `readme.md` to reflect that.

Reply:
In #1850, I implemented it such that the normalization is only applied to the targets, which is a problem if `shared_weights  == False`. It is not the same problem, but connected.